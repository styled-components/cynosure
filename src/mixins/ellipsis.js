// @flow
import type { Styles } from '../types/style'

/**
 * CSS to represent truncated text with an ellipsis.
 *
 * @example
 * // Styles as object usage
 * const styles = {
 *   ...ellipsis('250px')
 * }
 *
 * // styled-components usage
 * const div = styled.div`
 *   ${ellipsis('250px')}
 * `
 *
 * // CSS as JS Output
 *
 * div: {
 *   'display': 'inline-block',
 *   'display': 'inline-flex',
 *   'maxWidth': '250px',
 *   'overflow': 'hidden',
 *   'textOverflow': 'ellipsis',
 *   'whiteSpace': 'nowrap',
 *   'wordWrap': 'normal'
 * }
 */
function ellipsis(width?: string | number = '100%'): Array<Styles> {
  return [
    {
      display: 'inline-block',
      maxWidth: width,
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      wordWrap: 'normal',
    },
    {
      display: 'inline-flex',
    },
  ]
}

export default ellipsis
