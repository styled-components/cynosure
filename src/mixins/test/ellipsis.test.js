// @flow
import ellipsis from '../ellipsis'

describe('ellipsis', () => {
  it('should pass parameter to the value of max-width', () => {
    expect(ellipsis('300px')).toMatchSnapshot()
  })

  it('should pass parameter of type integer to the value of max-width', () => {
    expect(ellipsis(300)).toMatchSnapshot()
  })

  it('should default lines to 1 and max-width to 100%', () => {
    expect(ellipsis()).toMatchSnapshot()
  })

  it('should truncate text after 3 lines', () => {
    expect(ellipsis(null, 3)).toMatchSnapshot()
  })

  it('should truncate text after 3 lines and 500px max-width', () => {
    expect(ellipsis('500px', 3)).toMatchSnapshot()
  })
})
