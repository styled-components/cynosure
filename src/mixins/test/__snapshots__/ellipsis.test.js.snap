exports[`ellipsis should default max-width to 100% 1`] = `
Object {
  "display": "inline-block",
  "maxWidth": "100%",
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
  "wordWrap": "normal",
}
`;

exports[`ellipsis should pass parameter to the value of max-width 1`] = `
Object {
  "display": "inline-block",
  "maxWidth": "300px",
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
  "wordWrap": "normal",
}
`;

exports[`ellipsis should properly add rules when block has existing rules 1`] = `
Object {
  "background": "red",
  "display": "inline-block",
  "maxWidth": "300px",
  "overflow": "hidden",
  "textOverflow": "ellipsis",
  "whiteSpace": "nowrap",
  "wordWrap": "normal",
}
`;
